File: app/src/main/java/com/wmods/wppenhacer/xposed/AntiUpdater.java
package com.wmods.wppenhacer.xposed;

import android.content.pm.PackageInstaller;

import com.wmods.wppenhacer.xposed.core.FeatureLoader;

import java.io.IOException;

import de.robv.android.xposed.XC_MethodHook;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class AntiUpdater {

    public static void hookSession(XC_LoadPackage.LoadPackageParam lpparam) {
        if (lpparam.packageName.equals("android")) return;
        XposedBridge.hookAllMethods(PackageInstaller.class, "createSession", new XC_MethodHook() {
            @Override
            protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
                var session = (PackageInstaller.SessionParams) param.args[0];
                var packageName = XposedHelpers.getObjectField(session, "mPackageName");
                if (packageName.equals(FeatureLoader.PACKAGE_WPP) || packageName.equals(FeatureLoader.PACKAGE_BUSINESS)) {
                    param.setThrowable(new IOException("UPDATE LOCKED BY WAENHANCER"));
                }
            }
        });

    }
}

File: app/src/main/java/com/wmods/wppenhacer/xposed/AntiUpdater.kt
package com.wmods.wppenhacer.xposed

import android.content.pm.PackageInstaller
import com.wmods.wppenhacer.xposed.core.FeatureLoader
import de.robv.android.xposed.XC_MethodHook
import de.robv.android.xposed.XposedBridge
import de.robv.android.xposed.XposedHelpers
import de.robv.android.xposed.callbacks.XC_LoadPackage.LoadPackageParam
import java.io.IOException

object AntiUpdater {
    fun hookSession(lpparam: LoadPackageParam) {
        if (lpparam.packageName == "android") return
        XposedBridge.hookAllMethods(
            PackageInstaller::class.java,
            "createSession",
            object : XC_MethodHook() {
                override fun beforeHookedMethod(param: MethodHookParam) {
                    val session = param.args[0] as PackageInstaller.SessionParams?
                    val packageName = XposedHelpers.getObjectField(session, "mPackageName")
                    if (packageName == FeatureLoader.PACKAGE_WPP || packageName == FeatureLoader.PACKAGE_BUSINESS) {
                        param.setThrowable(IOException("UPDATE LOCKED BY WAENHANCER"))
                    }
                }
            })
    }
}
